From e490d91112ac8551fbb23d4e2ef58da57f2b5cf8 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Mon, 4 Jul 2022 17:28:20 +0800
Subject: [PATCH] gstreamer1.0-plugins-bad: CB1 the video will be stuck when
 switch segment [1/1]

PD#SWPL-85712

Problem:
When the firset fragmentdownload finish, will call advance fragmentonce,
Which will make the first fragment.timestamp raise up to next fragment
But at this time, the src_pad got a  eos from upstreaming, handle this eos event
Will raise up the  fragment.timestamp again, so the real downloading data is the second next framgment
Which will make video stuck

Solution:
Do not advance fragmentonce when downloading_header or downloading_index

Verify:
AP212
---
 ext/hls/gsthlsdemux.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/hls/gsthlsdemux.c b/ext/hls/gsthlsdemux.c
index d3ea6ee..9a6a62e 100644
--- a/ext/hls/gsthlsdemux.c
+++ b/ext/hls/gsthlsdemux.c
@@ -948,6 +948,9 @@ gst_hls_demux_finish_fragment (GstAdaptiveDemux * demux,
     }
   }
 
+  if (G_UNLIKELY (stream->downloading_header || stream->downloading_index))
+    return GST_FLOW_OK;
+
   gst_hls_demux_stream_clear_pending_data (hls_stream);
 
   if (ret == GST_FLOW_OK || ret == GST_FLOW_NOT_LINKED)
-- 
2.25.1

